src/python/pants/backend/python/target_types.py

```diff
@@ -167,6 +167,10 @@ class PythonRunGoalUseSandboxField(TriBoolField):
         If false, runs of this target with the `run` goal will use the in-repo sources
         directly.
 
+        Note that this field is going to apply only when running a target with the `run` goal.
+        Invoking the `test` goal on a `python_test` target with the field set to false won't
```

Comment:
I'm +1 the change the but wording is a bit awkward. 

Let me suggest some if no one else does, when I'm on my tower